src/expect.jl

```diff
@@ -1,57 +1,63 @@
-using ITensors.ITensorMPS: ITensorMPS, expect, promote_itensor_eltype, OpSum
+using Dictionaries: Dictionary, set!
+using ITensors: Op, op, contract, siteinds, which_op
+using ITensors.ITensorMPS: ITensorMPS, expect
+
+default_expect_alg() = "bp"
+
+function ITensorMPS.expect(ψIψ::AbstractFormNetwork, op::Op; contract_kwargs=(;), kwargs...)
```

Comment:
I've lost track, does this function ever run BP or does it just use the message tensors as they are?

Reply:
Depends. If you call it yourself and set `alg = "bp"` in the `kwargs` it will.

But when it gets called from `ITensorMPS.expect(ψ::AbstractITensorNetwork, op::Op; kwargs...)` to doesn't as that functions constructs the form network and does some cache logic before calling `ITensorMPS.expect(ψIψ::AbstractFormNetwork, op::Op; update_cache = false, kwargs...)`  internally

Reply:
I see, thanks, I missed the logic around `update_cache`, I'll take a closer look at that logic.